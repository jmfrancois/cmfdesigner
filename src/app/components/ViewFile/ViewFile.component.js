import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect, Inject } from '@talend/react-cmf';
import { Inject as PureInject } from '@talend/react-components';

const getReactElement = PureInject.getReactElement;

function ViewFile(props) {
	if (!props.file) {
		return null;
	}
	let title = props.title;
	if (!title) {
		title = props.file.path.split('/').pop();
	}
	return (
		<div>
			{props.onDelete && (
				<button
					className="btn btn-danger pull-right"
					onClick={event => props.onDelete(event, props.file)}
				>
					Delete
				</button>
			)}
			<h1>{title}</h1>
			{getReactElement(props.getComponent, props.content)}
			<Inject component="ViewLogs" file={props.file} />
			<Inject component="FileAnalytics" path={props.file.path} />
		</div>
	);
}

ViewFile.displayName = 'ViewFile';
ViewFile.propTypes = {
	onDelete: PropTypes.func,
	title: PropTypes.string,
	content: getReactElement.propTypes,
	file: PropTypes.shape({
		path: PropTypes.string.isRequired,
	}),
	...cmfConnect.propTypes,
};

export default cmfConnect({})(ViewFile);
