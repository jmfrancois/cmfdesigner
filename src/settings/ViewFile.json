{
  "props": {
    "ViewFile#logs": {
      "fileExpression": "service#designer.logs:getCurrent",
      "content": [
      ],
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/logs/view",
                "isExact": true
            }
        ]
      }
    },
    "ViewFile#component": {
      "titleExpression": "getComponentTitle",
      "fileExpression": "service#designer.components:getCurrent",
      "content": [
        { "component": "ViewComponent" }
      ],
      "onDeleteDispatchActionCreator": "service#designer#components:delete",
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/components/view",
                "isExact": true
            }
        ]
      }
    },
    "ViewFile#expression": {
      "titleExpression": "getExpressionTitle",
      "fileExpression": "service#designer.expressions:getCurrent",
      "content": [
        { "component": "ViewFunction", "fnExpression": "service#designer.expressions:getCurrent" }
      ],
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/expressions/view",
                "isExact": true
            }
        ]
      }
    },
    "ViewFile#props": {
      "titleExpression": "getPropsTitle",
      "fileExpression": "service#designer.props:getCurrent",
      "content": [
        { "component": "ObjectViewer", "componentId": "view-props", "id": "view-props", "data": {} }
      ],
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/props/view",
                "isExact": true
            }
        ]
      }
    },
    "ViewFile#saga": {
      "titleExpression": "getSagaTitle",
      "fileExpression": "service#designer.sagas:getCurrent",
      "content": [
        { "component": "ViewFunction", "fnExpression": "service#designer.sagas:getCurrent" }
      ],
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/sagas/view",
                "isExact": true
            }
        ]
      }
    }
  }
}
