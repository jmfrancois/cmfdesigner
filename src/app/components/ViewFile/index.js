import ViewFile from './ViewFile.component';

export default ViewFile;
