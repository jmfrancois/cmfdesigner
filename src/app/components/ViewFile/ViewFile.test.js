import React from 'react';
import { shallow } from 'enzyme';

import ViewFile from './ViewFile.component';

describe('ViewFile', () => {
	it('should render', () => {
		const wrapper = shallow(
			<ViewFile.WrappedComponent />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
