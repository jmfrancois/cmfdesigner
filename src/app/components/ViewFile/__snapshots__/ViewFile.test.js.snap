// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewFile should render 1`] = `null`;
